### PR TITLE
add jenkins and elasticsearch to backup docs

### PR DESCRIPTION
**Pull request by wallnerryan**
- add jenkins and elasticsearch to backup docs
- Update titles for cassandra and mysql

### CODE REVIEW

File: content/use-px-backup/backup-stateful-applications/elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+You can use the instructions on this page to create pre and post backup rules with PX-Backup, which take application-consistent backups for Elasticsearch on Kubernetes in production.
+
+You should first configure the Elasticsearch data directory tp use a PVC to prevent permanent data loss. Mount a Portworx volume for this purpose to `/user/share/elasticsearch/data` inside the Kubernetes pod. This will also enable PX-Backup to backup and restore the data stored in this location.
+
+Before usnig this guide, make sure and configure PVCs for `elasticsearch-data`. Use the below file as an example. 
+
+{{<info>}}
+**NOTE:** 
+
+The below template can not be used alone. Please follow pre-requisits from the following [elastic on kubernetes operations guide](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-eck.html).
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/elasticsearch.md:19]:
```suggestion
The template below can not be used alone. Please follow pre-requisites from the following [elastic on kubernetes operations guide](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-eck.html).
```

File: content/use-px-backup/backup-stateful-applications/cassandra.md [hunk 2]
```diff
@@ -1,14 +1,13 @@
 ---
-title: Cassandra
+title: Bacckup Cassandra on Kubernetes
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/cassandra.md:2]:
```suggestion
title: Backup Cassandra on Kubernetes
```

File: content/use-px-backup/backup-stateful-applications/elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+weight: 4
+hidesections: true
+disableprevnext: true
+---
+
+You can use the instructions on this page to create pre and post backup rules with PX-Backup, which take application-consistent backups for Elasticsearch on Kubernetes in production.
+
+You should first configure the Elasticsearch data directory tp use a PVC to prevent permanent data loss. Mount a Portworx volume for this purpose to `/user/share/elasticsearch/data` inside the Kubernetes pod. This will also enable PX-Backup to backup and restore the data stored in this location.
+
+Before usnig this guide, make sure and configure PVCs for `elasticsearch-data`. Use the below file as an example.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/elasticsearch.md:14]:
```suggestion
Before using this guide, make sure and configure PVCs for `elasticsearch-data`. Use the below file as an example. 
```

**Review by nathan-px — CHANGES_REQUESTED** [review]
typos, will review again and in more detail.

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+description: 
+keywords: backup, jenkins
+weight: 4
+hidesections: true
+disableprevnext: true
+---
+
+You can use the instructions on this page to create pre and post backup rules with PX-Backup, which take application-consistent backups for Jenkins on Kubernetes in production.
+
+Jenkins stores data within a directory known as `JENKINS_HOME`. You should configure a persistent volume for Jenkins so that the entire directory is backed by a volume to capture all data associated with Jenkins. Workspaces, plug-ins, jobs, user content and overall configuration within this directory is vital when recovering from various types of failures.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:12]: same note about prerequisites as the other article.

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+## Installation
+
+### Prerequisites
+
+{{% content "shared/common-info.md" %}}
+
+{{<info>}}
+**NOTE:** 
+These examples assume that the Jenkins CLI is available within the Jenkins pod. You can make sure its available in the Jenkins pod(s) by running the below command.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:38]:
```suggestion
**NOTE:** These examples assume that the Jenkins CLI is available within the Jenkins pod. You can make sure its available in the Jenkins pod(s) by running the below command.
```

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+Once you configure a PVC to be mounted at `/var/jenkins_home` you can use the below guide for pre and post backup rules.
+
+## Installation
+
+### Prerequisites
+
+{{% content "shared/common-info.md" %}}
+
+{{<info>}}
+**NOTE:**
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:37]:
```suggestion

```

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+Create rules for Jenkins that will run both before and after the backup operation runs:
+
+#### Create a pre-exec backup rule for Jenkins
+
+Let's assume the Jenkins deployment has a job called `job-1`. We will use the pre-exec rule to make sure this job is stopped before taking our snapshot.
+
+{{<info>}}
+**NOTE:** 
+Stopping a job is not necessary to backup Jenkins. It is used as an example of a pre backup rule. You may run other commands available within the Jenkins pod here as well.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:56]:
```suggestion
**NOTE:** Stopping a job is not necessary to backup Jenkins. It is used as an example of a pre backup rule. You may run other commands available within the Jenkins pod here as well.
```

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+{{</info>}}
+
+### Create rules for Jenkins
+
+Create rules for Jenkins that will run both before and after the backup operation runs:
+
+#### Create a pre-exec backup rule for Jenkins
+
+Let's assume the Jenkins deployment has a job called `job-1`. We will use the pre-exec rule to make sure this job is stopped before taking our snapshot.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:52]:
```suggestion
Assume the Jenkins deployment has a job called `job-1`, you will use the pre-exec rule to make sure this job is stopped before taking our snapshot.
```

File: content/use-px-backup/backup-stateful-applications/jenkins.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+	```text
+	java -jar /var/jenkins_home/cli.jar -s http://jenkins:8080 -webSocket -auth username:password stop-builds job-1
+	```
+
+    ![](/img/jenkins-pre-rule.png)
+
+#### Create a post-exec backup rule for Jenkins
+
+After our backup is triggered, we can start the build for `job-1` again using a post-exec rule. This ensures our job is not running during the backup and is restarted after the backup data is captured.
```
Comment (nathan-px) [inline content/use-px-backup/backup-stateful-applications/jenkins.md:77]:
```suggestion
After your backup is triggered, you can start the build for `job-1` again using a post-exec rule. This ensures your job is not running during the backup and is restarted after the backup data is captured. 
```

**Review by nathan-px — CHANGES_REQUESTED** [review]

File: content/use-px-backup/backup-stateful-applications/elasticsearch.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+You can use the instructions on this page to create pre and post backup rules with PX-Backup, which take application-consistent backups for Elasticsearch on Kubernetes in production.
+
+You should first configure the Elasticsearch data directory tp use a PVC to prevent permanent data loss. Mount a Portworx volume for this purpose to `/user/share/elasticsearch/data` inside the Kubernetes pod. This will also enable PX-Backup to backup and restore the data stored in this location.
+
+Before using this guide, make sure and configure PVCs for `elasticsearch-data`. Use the below file as an example. 
+
+{{<info>}}
+**NOTE:** 
+
+The template below can not be used alone. Please follow pre-requisites from the following [elastic on kubernetes operations guide](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-eck.html).
```
[the remark this answers is not in the document]
Comment (wallnerryan) [reply content/use-px-backup/backup-stateful-applications/elasticsearch.md:19]: ok makes sense.

File: content/use-px-backup/backup-stateful-applications/elasticsearch.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+description: 
+keywords: backup, elasticsearch
+weight: 4
+hidesections: true
+disableprevnext: true
+---
+
+You can use the instructions on this page to create pre and post backup rules with PX-Backup, which take application-consistent backups for Elasticsearch on Kubernetes in production.
+
+You should first configure the Elasticsearch data directory tp use a PVC to prevent permanent data loss. Mount a Portworx volume for this purpose to `/user/share/elasticsearch/data` inside the Kubernetes pod. This will also enable PX-Backup to backup and restore the data stored in this location.
```
[the remark this answers is not in the document]
Comment (wallnerryan) [reply content/use-px-backup/backup-stateful-applications/elasticsearch.md:12]: right. good catch. This is more of a pre-req as this applies to generally running stateful services. I'll update to identify it as a pre-req configuration of elasticsearch.

**Comment by wallnerryan** [comment]
ping @nathan-px

**Comment by wallnerryan** [comment]
Ping @nathan-px

**Comment by nathan-px** [comment]
> Ping @nathan-px

There are still some unresolved suggestions. Do you want to proceed without those changes?

**Comment by wallnerryan** [comment]
@nathan-px all suggestions should be resolved.

> > Ping @nathan-px
> 
> @wallnerryan There are still some unresolved suggestions. Do you want to proceed without those changes?

Missed those that were collapsed. Should be good now.

**Review by nathan-px — APPROVED** [review]

**Pull request merged**